src/components/views/context_menus/MessageContextMenu.js

```diff
@@ -351,7 +351,7 @@ export default class MessageContextMenu extends React.Component {
             );
         }
 
-        if (isSent && this.state.canRedact) {
+        if (isSent && this.state.canRedact && mxEvent.getType() === 'm.room.message') {
```

Comment:
Please add a configuration parameter, allowRoomAdminsToRedactSystemMessages which defaults to true, and we override it to false in our web app config.json file.  Remember we want to push our changes to the matrix repo and have the default behaviour result in no functional change.